src/runtime_src/core/common/device.h

```diff
@@ -511,6 +511,28 @@ device_query(const std::shared_ptr<device>& device, Args&&... args)
   return boost::any_cast<typename QueryRequestType::result_type>(ret);
 }
 
+template <typename QueryRequestType>
+inline typename QueryRequestType::result_type
+device_query_default(const device* device, const typename QueryRequestType::result_type& default_value)
+{
+  try {
+    return device_query<QueryRequestType>(device);
+  } catch(...) {
```

Comment:
```suggestion
  } 
  catch(const query::exception& ) {
```
Do not catch all exceptions here which could be unrelated to missing query implementation.  Maybe even the `query::exception` base class is too broad, but it's far better than `...`.